pydicer/convert/convert_pt.py

```diff
@@ -0,0 +1,311 @@
+                scan_datetime = datetime.strptime(scan_datetime_str, "%Y%m%d%H%M%S")
+
+            start_time_str = s.RadiopharmaceuticalInformationSequence[
+                0
+            ].RadiopharmaceuticalStartTime
+            start_time = time(
+                int(start_time_str[0:2]),
+                int(start_time_str[2:4]),
+                int(start_time_str[4:6]),
+            )
+            start_datetime = datetime.combine(scan_datetime.date(), start_time)
+            decay_time = (scan_datetime - start_datetime).total_seconds()
+        except KeyError:
+            warnings.warn(
+                "Estimation of time decay for SUV"
+                " computation from average parameters"
+            )
+            decay_time = 1.75 * 3600  # From Martin's code
+        except AttributeError:
+            warnings.warn(
+                "'Dataset' object has no attribute 'RadiopharmaceuticalStartTime'"
+            )
+            decay_time = 1.75 * 3600  # From Shuchao's code
+        suv_results = get_suv_from_bqml(slices, decay_time, patient_weight, dtype=dtype)
+
+    elif units == "CNTS":
+        suv_results = get_suv_philips(slices, dtype=dtype)
+    else:
+        raise ValueError("The {} units is not handled".format(units))
+
+    return suv_results
+
+
+def get_suv_philips(slices, dtype=np.float32):
+    """Function to Get SUV from raw PET if units = "CNTS"
+    Args:
+    Returns:
+    """
+    image = list()
+    suv_scale_factor_tag = Tag(0x70531000)
+    for s in slices:
+        if (
+                suv_scale_factor_tag in s
+                and "RescaleSlope" in s
+                and "RescaleIntercept" in s
+            ):  # # From Shuchao's code
+            im = (
+                float(s.RescaleSlope) * s.pixel_array + float(s.RescaleIntercept)
+            ) * float(s[suv_scale_factor_tag].value)
+        else:
+            im = (1.0 * s.pixel_array + 0.0) * 0.000587
+
+        image.append(im)
+
+    return np.stack(image, axis=-1).astype(dtype)
+
+
+def get_suv_from_bqml(slices, decay_time, patient_weight, dtype=np.float32):
+    """Function to Get SUV from raw PET if units = "BQML"
+    Args:
+    Returns:
+    """
+    image = list()
+    for s in slices:
+        pet = float(s.RescaleSlope) * s.pixel_array + float(s.RescaleIntercept)
+        if "RadionuclideHalfLife" in s.RadiopharmaceuticalInformationSequence[0]:
+            half_life = float(
+                s.RadiopharmaceuticalInformationSequence[0].RadionuclideHalfLife
+            )
+        else:
+            half_life = 6500.0  # # From Shuchao's code (default)
+            print("there is no RadionuclideHalfLife")
+        if "RadionuclideTotalDose" in s.RadiopharmaceuticalInformationSequence[0]:
+            total_dose = float(
+                s.RadiopharmaceuticalInformationSequence[0].RadionuclideTotalDose
+            )
+        else:
+            total_dose = 487254240.0  # # From Shuchao's code (default)

```

Comment:
Where do these default values come from? This seems like using a default value will give you some output but the numbers could be way off their true values? If this is the case, I think we should really make sure the code fails loudly here, rather than estimate some (probably incorrect) numbers. Since the user likely won't see this print message and will go on using the converted data as is.

We can discuss this with the group in Friday's meeting.

Reply:
Yes I just re-read your initial post. I see yes I think this requires more discussion. We can discuss it with the group and determine a good solution to this.